platform/src/main/java/org/stellar/anchor/platform/service/MetricEmitterService.java

```diff
@@ -0,0 +1,63 @@
+package org.stellar.anchor.platform.service;
+
+import io.micrometer.core.instrument.Metrics;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import javax.annotation.PreDestroy;
+
+import io.micrometer.core.instrument.Tags;
+import org.stellar.anchor.platform.data.JdbcSep31TransactionStore;
+import org.stellar.anchor.util.Log;
+
+public class MetricEmitterService {
+  private final ScheduledExecutorService executor = Executors.newScheduledThreadPool(1);
+  private final JdbcSep31TransactionStore sep31TransactionStore;
+  AtomicInteger pendingStellarTxns = new AtomicInteger(0);
+  AtomicInteger pendingCustomerInfoUpdateTxns = new AtomicInteger(0);
+  AtomicInteger pendingSenderTxns = new AtomicInteger(0);
+  AtomicInteger pendingReceiverTxns = new AtomicInteger(0);
+  AtomicInteger pendingExternalTxns = new AtomicInteger(0);
+  AtomicInteger completedTxns = new AtomicInteger(0);
+
+  public MetricEmitterService(JdbcSep31TransactionStore sep31TransactionStore) {
+    this.sep31TransactionStore = sep31TransactionStore;
+    this.executor.scheduleAtFixedRate(new MetricEmitter(), 0, 30, TimeUnit.SECONDS);
+    // create gauges for SEP-31 Transactions
+    Metrics.gauge("sep31.transaction", Tags.of("status", "pendingStellar"), pendingStellarTxns);
+    Metrics.gauge("sep31.transaction", Tags.of("status", "pendingCustomerInfoUpdate"), pendingCustomerInfoUpdateTxns);
+    Metrics.gauge("sep31.transaction", Tags.of("status", "pendingSender"), pendingSenderTxns);
+    Metrics.gauge("sep31.transaction", Tags.of("status", "pendingReceiver"), pendingReceiverTxns);
+    Metrics.gauge("sep31.transaction", Tags.of("status", "pendingExternal"), pendingExternalTxns);
+    Metrics.gauge("sep31.transaction", Tags.of("status", "completed"), completedTxns);
+
+    // TODO add gauges for SEP-24 Transactions
+
+  }
+
+  class MetricEmitter implements Runnable {
+    public void run() {
```

Comment:
These queries are quite expensive in JDBC. These are usually taken care of in analytics rather than metrics. 

Here is a short introduction to the difference between metrics and analytics. https://infogram.com/blog/metrics-vs-analytics/

People usually send logs to search engine such as ElasticSearch to perform transaction analytics.

Reply:
as discussed last week with @lijamie98 and @reecexlm, the metrics that require database queries will be configurable and will be disabled by default

Reply:
💡 Since this is going to be a recurring query, we should index the `status` column. Here's some background on that: https://www.baeldung.com/jpa-indexes